physicalplan/plan.go

```diff
@@ -66,9 +66,18 @@ func newOperator(expr parser.Expr, storage storage.Queryable, mint, maxt time.Ti
 				operators = append(operators, exchange.NewConcurrent(selector, 2))
 			}
 			return exchange.NewCoalesce(model.NewVectorPool(stepsBatch), operators...), nil
+		case *parser.NumberLiteral:
```

Comment:
Do we have an example of when this can happen?

Reply:
Yup, the example can be anything like `vector(1)` or PromQL functions which have scalar input. This is also present as a testcase. 🙂